pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/TopicPolicies.java

```diff
@@ -1814,4 +1815,102 @@ SchemaCompatibilityStrategy getSchemaCompatibilityStrategy(String topic, boolean
     CompletableFuture<Void> removeEntryFiltersPerTopicAsync(String topic);
 
 
+    /**
+     * Sets the autoSubscriptionCreation policy for a given topic, overriding namespace settings.
+     * <p/>
+     * When autoSubscriptionCreationOverride is enabled, new subscriptions will be created upon connection,
+     * regardless of the namespace level configuration.
+     * <p/>
+     * Request example:
+     *
+     * <pre>
+     * <code>
+     *  {
+     *      "allowAutoSubscriptionCreation" : true
+     *  }
+     * </code>
+     * </pre>
+     *
+     * @param topic
+     *            Topic name
+     * @param autoSubscriptionCreationOverride
+     *            Override policies for auto subscription creation
+     *
+     * @throws PulsarAdminException.NotAuthorizedException
+     *             Don't have admin permission
+     * @throws PulsarAdminException.NotFoundException
+     *             Topic does not exist
+     * @throws PulsarAdminException
+     *             Unexpected error
+     */
+    void setAutoSubscriptionCreation(
```

Comment:
I don't get the point of using `AutoSubscriptionCreationOverride` in the API. Isn't a `Boolean` more simple?

Reply:
In fact, it references the implementation of the namespace, as below:
https://github.com/apache/pulsar/blob/f029757ecb9653a97710f2bac563d247d04feb63/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java#L685-L701

Reply:
I see, it's introduced years ago in https://github.com/apache/pulsar/pull/6637.
It's too complex to expose this to admin API, but it's not easy to change it now.

Anyway, it's ok keep the consistency to me.